src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java

```diff
@@ -205,22 +205,56 @@ public void testGenerateFeedbackSessionEmails() throws IOException {
         
         ______TS("feedback session submission email");
 
-        Calendar time = Calendar.getInstance();
-        time.set(Calendar.DATE, 4);
-        time.set(Calendar.MONTH, 8);
-        time.set(Calendar.HOUR_OF_DAY, 5);
-        time.set(Calendar.MINUTE, 30);
+        Calendar timeForStudent = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        timeForStudent.set(Calendar.DATE, 4);
+        timeForStudent.set(Calendar.MONTH, 8);
+        timeForStudent.set(Calendar.HOUR_OF_DAY, 5);
+        timeForStudent.set(Calendar.MINUTE, 30);
         EmailWrapper email = new EmailGenerator().generateFeedbackSubmissionConfirmationEmailForStudent(
-                session, student1, TimeHelper.formatTime12H(time.getTime()));
+                session, student1, timeForStudent);
         subject = String.format(EmailType.FEEDBACK_SUBMISSION_CONFIRMATION.getSubject(), course.getName(),
                                 session.getFeedbackSessionName());
         verifyEmail(email, student1.email, subject, "/sessionSubmissionConfirmationEmailForStudent.html");
 
+        Calendar timeForInstructor = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        timeForInstructor.set(Calendar.DATE, 4);
+        timeForInstructor.set(Calendar.MONTH, 8);
+        timeForInstructor.set(Calendar.HOUR_OF_DAY, 5);
+        timeForInstructor.set(Calendar.MINUTE, 30);
         email = new EmailGenerator().generateFeedbackSubmissionConfirmationEmailForInstructor(session,
-                instructor1, TimeHelper.formatTime12H(time.getTime()));
+                instructor1, timeForInstructor);
         subject = String.format(EmailType.FEEDBACK_SUBMISSION_CONFIRMATION.getSubject(), course.getName(),
                                 session.getFeedbackSessionName());
         verifyEmail(email, instructor1.email, subject, "/sessionSubmissionConfirmationEmailForInstructor.html");
+        
+        ______TS("feedback session submission email with Singapore timezone");
+
+        session.setTimeZone(12.75);
+        Calendar timeForStudentSingapore = Calendar.getInstance(TimeZone.getTimeZone("GMT+08:00"));
+        timeForStudentSingapore.set(Calendar.DATE, 4);
+        timeForStudentSingapore.set(Calendar.MONTH, 8);
+        timeForStudentSingapore.set(Calendar.HOUR_OF_DAY, 10);
+        timeForStudentSingapore.set(Calendar.MINUTE, 30);
+        EmailWrapper emailSingapore = new EmailGenerator()
+                .generateFeedbackSubmissionConfirmationEmailForStudent(
+                        session, student1, timeForStudentSingapore);
+        subject = String.format(EmailType.FEEDBACK_SUBMISSION_CONFIRMATION.getSubject(), course.getName(),
+                session.getFeedbackSessionName());
+        verifyEmail(emailSingapore, student1.email, subject,
+                "/sessionSubmissionConfirmationEmailForStudentServerSingapore.html");
```

Comment:
You do not need to duplicate the emails because it's the same for both student and instructor; earlier it was done for coverage purpose. Now since there are multiple test case, we can do it this way:
- Test case 1: for student in positive timezone
- Test case 2: for instructor in negative timezone
- Test case 3: for student/instructor (free to choose) in zero timezone

You may need to rename the email result files.